aiogram/api/client/bot.py

```diff
@@ -98,15 +111,92 @@
     UserProfilePhotos,
     WebhookInfo,
 )
-from .base import BaseBot
+from ...utils.mixins import ContextInstanceMixin
+from ...utils.token import (
+    validate_token,
+    extract_bot_id,
+)
+
+T = TypeVar("T")
 
 
-class Bot(BaseBot):
+class Bot(ContextInstanceMixin["Bot"]):
```

Comment:
Just add at top of the file:
```python3
from __future__ import annotations
```
And then do:
```suggestion
class Bot(ContextInstanceMixin[Bot]):
```

Reply:
Nope:
```python
from __future__ import annotations

from typing import (
    TypeVar,
    Generic,
)

T = TypeVar("T")


class Genericable(Generic[T]):
    pass

class Quotes(Genericable[Quotes]):
    pass
```

```bash
>>> python a.py
NameError: name 'Quotes' is not defined
```